### PR TITLE
coded tcp server ... requires validation of chat room functionality

### PR DESCRIPTION
**Pull request by mizutombo**
Initial submission of 'tcp-chat-server' assignment.
Actions still required:
* test & validate functionality of chat room via telnet
* enable assignment of random nickname ID to client's message

### CODE REVIEW

**Comment by mizutombo** [comment]
resubmitting ...
debugged & validated chat
added random client ID feature

**Comment by mizutombo** [comment]
10/26 07:26
* Coded tcp tests for initial submission against LAB2 assignment.
* Debug in progress ... mocha just keeps running my original 'test.js' file instead of the new 'tcp-server-test.js' file.

**Comment by GeoffreyEmerson** [comment]
☹️  PR should be to your named branch, not master.

**Comment by GeoffreyEmerson** [comment]
❌  Also fails linter. Please run your "npm test" before making a PR.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  "version": "1.0.0",
+  "description": "",
+  "main": "tcp.js",
+  "scripts": {
+    "pretest": "npm run lint",
+    "test": "npm run test:run",
+    "test:run": "mocha",
+    "test:watch": "npm run test:run -- -w",
+    "lint": "eslint"
+  },
```
Comment (GeoffreyEmerson) [inline package.json]: 🗒  Moving forward, please include a "start" script. Something like this:

`"start": "npm run test && node index.js"`

File: tcpChatRoom.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+// 3. track clients leaving and announce
+
+module.exports = class tcpChatRoom {
+  constructor() {
+    this.seed = 1;
+    this.clients = [];
+  }
+  add(client) {
+    client.id = 'chatter ' + (this.seed++);
+    client.write("welcome to the discussion, " + `${client.id}`);
```
Comment (GeoffreyEmerson) [inline tcpChatRoom.js]: String interpolation is a *replacement* for string concatenation. So this line would look something like this:

```
`welcome to the discussion, ${client.id}`
```

File: tcpChatRoom.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  }
+  send(sender, message) {
+    this.clients.forEach(client => {
+      // if client is same as sender, then exit this forEach
+      // to prevent sender from receiving their own message
+      if(client === sender) return;
+      // otherwise, send message to other clients in chat room
+      // randomize client ID before sending message
+      var randomID = Math.floor((Math.random()*10));
+      sender.id = 'chatter ' + randomID;
```
Comment (GeoffreyEmerson) [inline tcpChatRoom.js]: Why are you randomizing the sender id? Just curious.

**Pull request closed without merging**